### ISSUE TITLE
Success indicator

### ISSUE THREAD

**Issue by laughedelic** [steelbrain/busy-signal#22]
It would be nice to have some indicator of whether the long-running process finished with success or a failure (through an added flag to the `clear` or `remove` API). 

I know that in the case of linter, I will have non-zero errors-warnings-infos, but for a more general usage it would be better to have in busy-signal. For example the dot could turn red instead of green on failure (after spinning). Or checkmark/cross..

I mostly just want it to be obvious when something finished successfully, because sometimes the spinning icon blinks too fast and I don't know whether it worked at all, or sometimes another plugin using busy-signal doesn't parse errors correctly, so the build fails, but there is no indication of it.

**Comment by steelbrain** [comment]
Whelp, did you try hovering over the icon for a second? It shows a history of everything that's executed

If something fails, and the user should be notified of it, most packages show a notification or something and those who don't, should. But if you still think we need success/error indicators, can you please provide your use case?

Also how will busy signal know if the previous error has been resolved because multiple providers are gonna provide different statuses, some are gonna be successful, some are not. And because providers can add more than one status at a time, it confuses everything further.

**Comment by laughedelic** [comment]
> did you try hovering over the icon for a second? It shows a history of everything that's executed

Yes, but it doesn't say anything about success or failure. Besides, I don't like moving cursor there to hover and waiting that split second to see the history. I see that it can be useful for somebody, but even if it showed success/failure in that tooltip (as the other busy package does) it wouldn't be very useful for me.

> If something fails, and the user should be notified of it, most packages show a notification or something and those who don't, should

Right. In most cases, of course a package would use linter to indicate any errors. But what about success? I don't think that "absence" of errors is a good indicator of success. 

> But if you still think we need success/error indicators, can you please provide your use case?

I'm using [atom-sbt](https://github.com/inkytonik/atom-sbt) plugin, which has an integration with busy-signal. It doesn't have a direct communication with the build tool (sbt), but parses it's output and adds messages to the linter. Such a workflow is not 100% robust as you can imagine. So sometimes it's not really clear whether it all went fine (even if it doesn't show errors). So I want to see some explicit indicator when it _did_ finish successfully.

> Also how will busy signal know if the previous error has been resolved because multiple providers are gonna provide different statuses, some are gonna be successful, some are not. 

Well, I saw this API in the busy package: [`end(identifier[, success])`](https://github.com/noseglid/atom-busy#endidentifier-success). Although it shows it only in the tooltip.

> And because providers can add more than one status at a time, it confuses everything further.

I think that if you have several processes and at least one of them fails, the indicator shouldn't be successful. It could also have 3 states: green (success), yellow (some failures), red (everything failed). Or just two for simplicity: green and red. It's just an idea, probably you have a better one.

**Comment by steelbrain** [comment]
> Right. In most cases, of course a package would use linter to indicate any errors. But what about success? I don't think that "absence" of errors is a good indicator of success.

> I'm using atom-sbt plugin, which has an integration with busy-signal. It doesn't have a direct communication with the build tool (sbt), but parses it's output and adds messages to the linter. Such a workflow is not 100% robust as you can imagine. So sometimes it's not really clear whether it all went fine (even if it doesn't show errors). So I want to see some explicit indicator when it did finish successfully.

So if I read it correctly, the provider returns errors to linter if it has any, and doesn't if it doesn't have any. How would it know it if failed and then fail it's entry on `busy-signal`, and if it has a way of knowing it failed, then why doesn't it show it as a notification? Like ALL of standard linters do.

<img src="https://cloud.githubusercontent.com/assets/4278113/25315200/da8aff38-286a-11e7-9c3d-246d0492365e.png">

I don't mind adding the API as much, but I'm just trying to confirm the use case here. Thank you for being patient throughout this

**Comment by laughedelic** [comment]
No problem. 

> So if I read it correctly, the provider returns errors to linter if it has any, and doesn't if it doesn't have any. How would it know it if failed and then fail it's entry on busy-signal, and if it has a way of knowing it failed, then why doesn't it show it as a notification?

Right, so if atom-sbt parses any errors in the sbt's output, it adds them to the linter, but the problem is that it needs also a way to know when the build finished and there's no need to wait for more output. For that it waits for the next prompt of the sbt-shell, which is a bit tricky, because a user can change the pattern of the prompt. I want to change that, because before the prompt there is a message which tells whether the build was successful or not. I want to use that information to indicate it with busy-signal.

Anyway, this is probably too involved into details and issues of a particular case. I think that a general possibility of communicating this type of information to the user is valuable. Whether to use it or not is a decision of the one who makes the integration with busy signal. 

Also, I see it as a generally useful plugin, which is not necessarily tied to linter (especially with the feature in #15). It could show a result of running some independent command, which may finish with a success or a failure and it doesn't have to do anything with the code itself (a silly example from the top of my head: upload current buffer to a Github gist, you can use busy-signal to show the progress and indicate the result). Sure, you can use a general error/info bubble, but I think it's quite an annoying kind of UI which shouldn't be used in the case of casual "expected" failures.

P.S. By the way, there's a typo in that error message in the word "Toggle".

**Comment by lharper** [comment]
I think this feature (combined with the progress indicator https://github.com/steelbrain/busy-signal/issues/15) would be useful as well.

I use the remote-ftp package to sync files to a server and find all the popup windows to be pretty distracting so I was hoping to modify that package to use this one for logging the feedback messages and completion status in an unobtrusive manner.
